truthy-falsy.js

```diff
@@ -0,0 +1,219 @@
+  if (secondVariable && !firstVariable) {
+    return true;
+  }
+  return false;
+}
+console.log(xor(true, true));
+
+// 4
+function isTruthy(booleanVariable, isBooleanTruth, isBooleanFalse) {
+  if (booleanVariable === true) {
+    return isBooleanTruth();
+  }
+  return isBooleanFalse();
+}
+
+// 5
+function rentalCarCost(days) {
+  const baseCost = days * 40;
+  if (days >= 7) {
+    return baseCost - 50;
+  }
+  if (days >= 3) {
+    return baseCost - 20;
+  }
+  return days * 40;
+}
+
+// 6
+function getRealFloor(floorNumber) {
+  if (floorNumber <= 0) {
+    return floorNumber;
+  }
+  if (floorNumber >= 13) {
+    return floorNumber - 2;
+  }
+  return floorNumber - 1;
+}
+
+// 7
+function willRockWin(opponent) {
+  return opponent === 'lizard' || opponent === 'scissors';
+}
+function willPaperWin(opponent) {
+  return opponent === 'rock' || oponent === 'spock';
+}
+function willScissorsWin(opponent) {
+  return opponent === 'paper' || opponent === 'lizard';
+}
+function willLizardWin(opponent) {
+  return opponent === 'spock' || opponent === 'paper';
+}
+function willSpockWin(opponent) {
+  return opponent === 'scissors' || 'rock';
+}
+
+function rockPaperScissorsLizardSpock(playerOne, playerTwo) {
+  if (playerOne === playerTwo) {
+    return 'Draw!';
+  }
+  if (playerOne === 'rock' && willRockWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'paper' && willPaperWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'scissors' && willScissorsWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'lizard' && willLizardWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  if (playerOne === 'spock' && willSpockWin(playerTwo)) {
+    return 'Player 1 Won!';
+  }
+  return 'Player 2 Won!';
+}
+
+// 8
+function isValidPassword(password) {
+  const specialCharacters = /[`!@#$%^&*()_\-+=\[\]{};':"\\|,.<>\/?~ ]/;
+  const passwordLengthCheck = password.trim().length >= 8;
+  const passwordCapitalCharactersCheck = /[A-Z]/.test(password) >= 1;
+  const passwordSpecialCharactersCheck = specialCharacters.test(password) >= 1;
+
+  if (passwordLengthCheck && passwordCapitalCharactersCheck && passwordSpecialCharactersCheck) {
+    return true;
+  }
+  return false;
+}
+console.log(isValidPassword('passSword!'));
+
+// 9
+function isPasswordLengthValid(password) {
+  if (password.trim().length >= 8) {
+    return 1;
+  }
+  return 0;
+}
+console.log(isPasswordLengthValid('pass'));
+
+function hasUppercaseLetter(password) {
+  if (/[a-z]/.test(password) >= 1 ) {
+    return 1;
+  }
+  return 0;
+}
+
+function hasLowercaseLetter(password) {
+  if (/[A-Z]/.test(password) >= 1 ) {
+    return 1;
+  } return 0;
+}
+
+function hasSpecialLetter(password) {
+  const specialChars =/[`!@#$%^&*()_\-+=\[\]{};':"\\|,.<>\/?~ ]/;
+  if (specialChars.test(password) >= 1 ) {
+    return 1;
+  }
+  return 0;
+}
+
+function getPasswordRating(password) {
+  const passwordLength = isPasswordLengthValid(password);
```

Comment:
The name `passwordLength` suggests that this variable holds the length of the password. This is not the case